sdk/monitor/azquery/MIGRATION.md

```diff
@@ -0,0 +1,180 @@
+ 
+ There have been a number of [terminology](https://learn.microsoft.com/azure/azure-monitor/terminology) changes for Azure Monitor over the years which resulted in the operations being spread over multiple packages. For Go, metrics methods were contained in `github.com/Azure/azure-sdk-for-go/services/preview/monitor/mgmt/<version-number>/insights` and logs methods resided in `github.com/Azure/azure-sdk-for-go/services/operationalinsights/v1/operationalinsights`.
+
+The new `azquery` module condenses metrics and logs functionality into one package for simpler access. The `azquery` module contains two clients: LogsClient and MetricsClient.
+
+Transitioning to a single package has resulted in a number of name changes, as detailed below.
+
+### Log name changes
+
+| `operationalinsights`    | `azquery` |
+| ----------- | ----------- |
+| QueryClient.Execute      | LogsClient.QueryWorkspace     |
+| MetadataClient.Get and MetadataClient.Post | N/A |
+
+The `azquery` module does not contain the `MetadataClient`. For that functionality, please use the old `operationalinsights` module or [file an issue in our github repo](https://github.com/Azure/azure-sdk-for-go/issues), so we can prioritize adding it to `azquery`.
+
+### Metrics name changes 
+
+| `insights` | `azquery` |
+| ----------- | ----------- |
+| MetricsClient.List     | MetricsClient.QueryResource       |
+| MetricDefinitionsClient.List   | MetricsClient.NewListDefinitionsPager        |
+| MetricNamespacesClient.List   | MetricsClient.NewListNamespacesPager        |
+
+## Query Logs
+
+### `operationalinsights`
+```go
+    import (
+        "context"
+        "github.com/Azure/azure-sdk-for-go/services/operationalinsights/v1/operationalinsights"
+        "github.com/Azure/go-autorest/autorest"
+    )
+
+    // create the client
+    client := operationalinsights.NewQueryClient()
+
+    client.Authorizer = autorest.NewAPIKeyAuthorizerWithHeaders(map[string]interface{}{
+		"x-api-key": "DEMO_KEY",
+	})
+
+    // execute the query
+    query := "<kusto query>"
+    timespan := "2023-12-25/2023-12-26"
+
+    res, err := client.Execute(context.TODO(), "DEMO_WORKSPACE", operationalinsights.QueryBody{Query: &query, Timespan: &timespan})
+	if err != nil {
+		//TODO: handle error
+	}
+```
+
+### `azquery`
+
+Compared to previous versions, querying logs with the new `azquery` module is clearer and simpler. There are a number of name changes for clarity, like how the old `Execute` method is now `QueryWorkspace`. In addition, there is improved time support. Before if a user added a timespan over which to query the request, it had to be a string constructed in the ISO8601 interval format. Users frequently made mistakes when constructing this string. With the new `QueryWorkspace` method, the type of timespan has been changed from a string to a new type named `TimeInterval`. `TimeInterval` has a contructor that allows users to take advantage of Go's time package, allowing easier creation.
```

Comment:
I think you can do some quick pkg.go.dev links here too for QueryWorkspace and TimeInterval.

Reply:
fixed in 36b33f8